### PR TITLE
Add `PascalCase` option to the `filename-case` rule

### PR DESCRIPTION
**Pull request by gokaygurcan**
Hello,

I need to use **PascalCase** for one of my projects and I thought it'd be nice to have it in this repo. I know that there're tons of people who uses PascalCase for their class names or model names, etc. So yea, I hope it'll help. 

```bash
$ npm test

> eslint-plugin-xo@0.3.1 test /Users/gokaygurcan/Projects/GitHub/eslint-plugin-xo
> xo && nyc ava


   4 passed
----------------------|----------|----------|----------|----------|----------------|
File                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------------------|----------|----------|----------|----------|----------------|
 rules/               |    95.38 |     87.5 |      100 |    95.38 |                |
  catch-error-name.js |    90.32 |       80 |      100 |    90.32 |        8,14,18 |
  filename-case.js    |      100 |      100 |      100 |      100 |                |
  no-process-exit.js  |      100 |      100 |      100 |      100 |                |
  throw-new-error.js  |      100 |      100 |      100 |      100 |                |
----------------------|----------|----------|----------|----------|----------------|
All files             |    95.38 |     87.5 |      100 |    95.38 |                |
----------------------|----------|----------|----------|----------|----------------|
```

### CODE REVIEW

File: rules/filename-case.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var path = require('path');
 var camelCase = require('lodash.camelcase');
 var kebabCase = require('lodash.kebabcase');
 var snakeCase = require('lodash.snakecase');
+var pascalCase = require('pascal-case');
```
Comment (jamestalmage) [inline rules/filename-case.js]: This brings in a whole bunch of duplicate dependencies. Instead let's stick with lodash:

```js
var capitalize = require('lodash.capitalize');

var pascalCase = function(string) {
  return capitalize(camelCase(string));
};

**Comment by jamestalmage** [comment]
@gokaygurcan - No need to copy paste the test output. We have a CI server that runs the tests for every pull request.

File: docs/rules/filename-case.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ Enforces all linted files to have their names in a certain case style. Default i
 - `foo_bar.test.js`
 - `foo_bar.test_utils.js`
 
+### `PascalCase`
```
Comment (jfmengels) [inline docs/rules/filename-case.md]: This should start with a lowercase letter, like the name of the option

File: rules/filename-case.js (hunk 1 above)
Comment (gokaygurcan) [inline rules/filename-case.js]: Hmm, it was the one that I normally use but you're right. I'll update the deps and stick with Lodash only.

File: docs/rules/filename-case.md (hunk 2 above)
Comment (gokaygurcan) [inline docs/rules/filename-case.md]: Arghh! Fixing asap.

**Comment by jfmengels** [comment]
LGTM :)

**Comment by sindresorhus** [comment]
Thanks @gokaygurcan :)

**Comment by gokaygurcan** [comment]
Thank you all, for your kindness and suggestions. Now, I can use XO without any problem ;)

**Pull request merged**